src/database/operations/create/createProposal.ts

```diff
@@ -0,0 +1,18 @@
+import { db } from '../../db';
+import { NotFoundError } from '../../../errors';
+import type { Proposal, JsonResultSet, WritableProposal } from '../../../types';
+
+export const createProposal = async (
+	createValues: WritableProposal,
+): Promise<Proposal> => {
+	const { opportunityId, externalId } = createValues;
+	const result = await db.sql<JsonResultSet<Proposal>>('proposals.insertOne', {
+		opportunityId,
+		externalId,
+	});
+	const proposal = result.rows[0]?.object;
+	if (proposal === undefined) {
+		throw new NotFoundError('The proposal could not be created.');
```

Comment:
Oh good call, notfound feels incorrect!  I think you're right about 500 error.

And to answer your question.... this really is not a reachable state (which is why it's not tested) -- the only way to reach it would be if... the database did not error BUT... did not return a result set?  What?  I don't think that's possible.

To that end, maybe the right status code is `418 I'm a teapot`